lib/jekyll-seo-tag/drop.rb

```diff
@@ -0,0 +1,239 @@
+module Jekyll
+  class SeoTag
+    class Drop < Jekyll::Drops::Drop
+      include Jekyll::SeoTag::JSONLD
+
+      TITLE_SEPARATOR = " | ".freeze
+      FORMAT_STRING_METHODS = %i[
+        markdownify strip_html normalize_whitespace escape_once
+      ].freeze
+
+      def initialize(text, context)
+        @obj = {}
+        @mutations = {}
+        @text    = text
+        @context = context
+      end
+
+      def version
+        Jekyll::SeoTag::VERSION
+      end
+
+      # Should the `<title>` tag be generated for this page?
+      def title?
+        return false unless title
+        @text !~ %r!title=false!i
+      end
+
+      def site_title
+        @site_title ||= format_string(site["title"] || site["name"])
+      end
+
+      # Page title without site title or description appended
+      def page_title
+        @page_title ||= format_string(page["title"] || site_title)
+      end
+
+      # Page title with site title or description appended
+      def title
+        @title ||= begin
+          if page["title"] && site_title
+            page_title + TITLE_SEPARATOR + site_title
+          elsif site["description"] && site_title
+            site_title + TITLE_SEPARATOR + format_string(site["description"])
+          else
+            page_title || site_title
+          end
+        end
+      end
+
+      def name
+        @name ||= begin
+          return seo_name if seo_name
+          return unless homepage_or_about?
+
+          if site["social"] && site["social"]["name"]
+            format_string site["social"]["name"]
+          elsif site_title
+            format_string site_title
+          end
+        end
+      end
+
+      def description
+        @description ||= begin
```

Comment:
Is the begin here necessary? I believe you can keep this on 2 lines if needed without the `begin...end`